pkgs/build-support/setup-hooks/reproducible-builds.sh

```diff
@@ -1,4 +1,7 @@
 # Use the last part of the out path as hash input for the build.
 # This should ensure that it is deterministic across rebuilds of the same
 # derivation and not easily collide with other builds.
-export NIX_CFLAGS_COMPILE+=" -frandom-seed=${out##*/}"
+# We also truncate the hash so that it cannot cause reference cycles.
+local outbase="${out##*/}"
```

Comment:
Does local really just span this file? Setup hooks are sourced via stdenv, right?

Reply:
Ah I think you are right. How about putting this in a `getRandomSeed` function? I can't just inline it the basename + trimming because bash won't do the nested expansion

Reply:
Ok, I ended up using a command substitution. Looks a little odd but avoids any namespace pollution